android/modules/network/src/java/ti/modules/titanium/network/socket/TCPProxy.java

```diff
@@ -422,22 +421,51 @@ public int read(Object args[]) throws IOException
 			throw new IllegalArgumentException("Invalid number of arguments");
 		}
 
-		if (inputStream == null) {
```

Comment:
I assume we had this null guard for a good reason? We do retain the input stream as a field here, so I think we should still be re-using/assigning to it